cfme/middleware/domain.py

```diff
@@ -124,8 +125,7 @@ def _domains_in_mgmt(cls, provider):
         rows = provider.mgmt.inventory.list_domain()
         for row in rows:
             domains.append(MiddlewareDomain(
-                name=row.data['Local Host Name']
-                if row.data['Local Host Name'] != 'master' else 'Unnamed Domain',
+                name=re.sub(r'master\.', '', re.sub(r'%20', ' ', row.path.feed_id)),
```

Comment:
OPTIONAL: I don't think you need re.sub here. You can do this with ordinary string's "replace" method.